<?xml version="1.0"?>

<robot name="wrist" xmlns:xacro="http://ros.org/wiki/xacro">

    <xacro:macro name="wrist" params="prefix version">

        <link
                name="${prefix}wrist_base">

            <visual>
                <origin
                        xyz="0 0 0"
                        rpy="0 0 0"/>
                <geometry>
                        <mesh filename="package://festo_phand_description/meshes/wrist/wrist_base.STL"/>
                </geometry>


                     <xacro:black />

            </visual>
            <collision>
                <origin
                        xyz="0 0 0"
                        rpy="0 0 0"/>
                <geometry>
                     <mesh filename="package://festo_phand_description/meshes/wrist/wrist_base.STL"/>
                </geometry>
            </collision>
        </link>

        <link
                name="${prefix}wrist_top">

            <visual>
                <origin
                        xyz="0 0 0"
                        rpy="0 0 0"/>
                <geometry>
                    <mesh filename="package://festo_phand_description/meshes/wrist/wrist_top.STL"/>
                </geometry>

                    <xacro:black />

            </visual>
            <collision>
                <origin
                        xyz="0 0 0"
                        rpy="0 0 0"/>
                <geometry>
                    <mesh filename="package://festo_phand_description/meshes/wrist/wrist_top.STL"/>
                </geometry>
            </collision>
        </link>

        <link
                name="${prefix}mid">


        </link>

        <joint
                name="rotation_x"
                type="revolute">
            <origin
                    xyz="0 0 0"
                    rpy="0 0 0"/>

            <parent link="${prefix}wrist_base"/>

            <child link="${prefix}mid"/>

            <axis
                    xyz="1 0 0"/>

                <limit
                    lower="-0.174"
                    upper="0.174"
                    effort="0"
                    velocity="4"/>
        </joint>

         <joint
                name="rotation_y"
                type="revolute">
            <origin
                    xyz="0 0 0"
                    rpy="0 0 0"/>

            <parent link="${prefix}mid"/>

            <child link="${prefix}wrist_top"/>

            <axis
                    xyz="0 1 0"/>

                <limit
                    lower="-0.174"
                    upper="0.174"
                    effort="0"
                    velocity="4"/>
        </joint>

    </xacro:macro>
</robot>